content/usgs-packages/Application_Part4.Rmd

```diff
@@ -0,0 +1,268 @@
+}
+```
+
+## Create location to publish on SB
+
+The challenge with this application was to provide summaries of precipitation, nitrogen, and phosphorus data for a specific set of sites provided by a cooperator. The challenge was to automate the entire workflow, so the final step is to publish our results to ScienceBase. It would make most since to push the results to the same ScienceBase item that the cooperator provided for sites, but since this is an exercise and others will be completing it, we will save the results to a personal SB item. 
+
+Therefore, the first step is to create a folder under your user to save the results. Title this new item "usgs-pkgs-application-results". Visit the [sbtools modify lesson](/usgs-packages/sbtools-modify) to remind yourself how to do this. Try it on your own before expanding the solution code!
+
+```{r create-new-item, addToggle="ON"}
+# automatically created under the authenticated user
+sb_results_item <- item_create(title = "usgs-pkgs-application-results")
+
+# you would only create the item once, then you could just use its id moving forward
+sb_results_id <- sb_results_item$id
+```
+
+## Publish plots as images
+
+Next save the graphics as PNG files and upload to ScienceBase (add this code to the loop where the plots are rendered). 
+
+```{r automate-plots-publish, addToggle="ON"}
+site_fnames <- paste0("timeseries_", sites, ".png")
+
+for(i in seq_along(sites)){
+  site_i <- sites[i]
+  huc_site_i <- filter(sb_sites_info, site_no == site_i)$huc_cd # corresponding HUC8
+
+  precip_site_i <- filter(precip_data_long, huc8 == huc_site_i)
+  nitrogen_site_i <- filter(nitrogen_data, site_no == site_i)
+  phosphorus_site_i <- filter(phosphorus_data, site_no == site_i)
+  
+  png(filename = site_fnames[i], width=8, height=5, units="in", res=100)
+  
+  layout(matrix(1:3, nrow=3))
+  plot(precip_site_i$DateTime, precip_site_i$precip,
+       col="red", pch=20, xlab = "Time", ylab = "Precip accumulation, in",
+       main = paste("Site", site_i))
+  plot(nitrogen_site_i$sample_dt, nitrogen_site_i$result_va, 
+       col="green", pch=20, xlab = "Time", ylab = "Nitrogren concentration, mg/l")
+  plot(phosphorus_site_i$sample_dt, phosphorus_site_i$result_va,
+       col="blue", pch=20, xlab = "Time", ylab = "Phosphorus concentration, mg/l")
+  
+  dev.off()
+}
+
+updated_item <- item_append_files(sb_results_id, files = site_fnames)
+rm_files <- file.remove(site_fnames) # now that it's online, remove local copy
+```
+
+Check to see if the files were successfully uploaded to SB:
+
+```{r verify-upload}
+sb_fnames <- item_list_files(sb_results_id)
+all(site_fnames %in% sb_fnames$fname)
+```
+
+## Completed modular workflow
+
+You should now have a complete, modular workflow - from data retrieval to processing to visualizing, and finally to publishing and sharing. You could change the ScienceBase item ID to point to different site numbers to use for other cooperator's data. See the completed workflow below. 
```

Comment:
"You should now" -> "You now"
"other cooperator's" -> "another cooperator's" (or "other cooperators'")